packages/docs/docs/core-api/user-state.md

```diff
@@ -67,10 +67,10 @@ state.getEpochKeys(epoch: number): bigint[]
 
 ## getRepByAttester
 
-Get the reputation balance for an attester in an epoch. By default we use the currently specified attester and the latest epoch.
+Get the reputation balance for the current attester up to an epoch. By default we use the epoch before the current epoch. Reputation received in the current epoch isn't considered final until the user executes a user state transition out of the epoch.
```

Comment:
`epoch before the current epoch` sounds like `epoch = currentEpoch - 1`, is it correct?
Or we can be more specific: `By default we use the latest user transition epoch before the current epoch`

Reply:
> epoch before the current epoch sounds like epoch = currentEpoch - 1, is it correct?
Yes

> By default we use the latest user transition epoch before the current epoch

This isn't correct, if the current epoch is 5, and the user last transitioned to epoch 1, we'll show the reputation up until epoch 4.

Reply:
The reason we do this is because some other internal functions need to get attestations from `currentEpoch - 1` to do things like epoch transition. I think it might be better to by default show the balance up to `currentEpoch`, then in the internal functions explicitly specify `currentEpoch - 1`

Reply:
> This isn't correct, if the current epoch is 5, and the user last transitioned to epoch 1, we'll show the reputation up until epoch 4.

I think it takes the `latestTransitionedEpoch`, and it finds the `epkNullifier` from db
so if the user doesn't perform UST, the nullifier cannot be found

https://github.com/Unirep/Unirep/blob/68c293ce216e15e73395f30b80984d0cc1335775/packages/core/src/UserState.ts#L72

and I think it doesn't make sense if user doesn't perform UST but receive reputation 😧
that means other users/attesters can predict epoch keys and give reputation

Reply:
But I think in this function `getRepByAttester` it is fine to query from `currentEpoch - 1`
and this line should be changed
https://github.com/Unirep/Unirep/blob/68c293ce216e15e73395f30b80984d0cc1335775/packages/core/src/UserState.ts#L191

Reply:
Right `latestTransitionedEpoch` is the default value it processes up to. And we use a `<` operator instead of `<=` so if the `currentEpoch` is passed to the function we process up until `currentEpoch - 1`. We just need to change the default value and check where else it's used and change `<` to `<=`. Then if the user passes `currentEpoch` they'll get the balance up to and including the current epoch.